authentication/authentication_test.go

```diff
@@ -0,0 +1,125 @@
+package authentication
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	_ "github.com/joho/godotenv/autoload"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/auth0/go-auth0/authentication/database"
+)
+
+var (
+	domain                = os.Getenv("AUTH0_DOMAIN")
+	clientID              = os.Getenv("AUTH0_AUTH_CLIENT_ID")
```

Comment:
Could the env variable be simply `AUTH0_CLIENT_ID`? Considering we have `AUTH0_DOMAIN` above.

Reply:
When I was testing my client id I used for the management tests wouldn't work here, I'm not sure if that was just down to me not having that client configure correctly to do auth and management calls or not. I'll investigate further and potentially stick to one env var as suggested

Reply:
I enabled all grants for the M2M app used for management tests and it doesn't seem to work for the `TestOAuthLoginWithPassword` test (however it does work for signup), I think for now it'll be best to keep the split and see if this can be improved in future.